File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3-coder-30b-mlx_nothink/20260830_231336/typescript/001_csv_to_json/solution.ts
import { readFileSync } from 'fs';
import { join } from 'path';

const filePath = join('input', 'data.csv');
const fileContent = readFileSync(filePath, 'utf-8');

const lines = fileContent.trim().split('\n');
const headers = lines[0].split(',');

const result = lines.slice(1).map(line => {
  const values = line.split(',');
  const obj: any = {};
  
  headers.forEach((header, index) => {
    const key = header.trim();
    const value = values[index]?.trim();
    
    switch (key) {
      case 'Name':
        obj[key] = value;
        break;
      case 'Age':
        obj[key] = parseInt(value, 10);
        break;
      case 'Email':
        obj[key] = value;
        break;
      case 'Score':
        obj[key] = parseFloat(value);
        break;
    }
  });
  
  return obj;
});

console.log(JSON.stringify(result));

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3-coder-30b-mlx_nothink/20260830_231336/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"io"
	"os"
	"strconv"
	"strings"
)

type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	file, err := os.Open("input/data.csv")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	reader := csv.NewReader(file)
	var people []Person

	// Read and skip the header row
	_, err = reader.Read()
	if err != nil {
		panic(err)
	}

	for {
		record, err := reader.Read()
		if err == io.EOF {
			break
		}
		if err != nil {
			panic(err)
		}

		age, err := strconv.Atoi(record[1])
		if err != nil {
			panic(err)
		}

		score, err := strconv.ParseFloat(record[3], 64)
		if err != nil {
			panic(err)
		}

		person := Person{
			Name:  strings.TrimSpace(record[0]),
			Age:   age,
			Email: strings.TrimSpace(record[2]),
			Score: score,
		}

		people = append(people, person)
	}

	jsonData, err := json.Marshal(people)
	if err != nil {
		panic(err)
	}

	fmt.Println(string(jsonData))
}